streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -62,24 +74,31 @@ public void addChild(ProcessorNode<?, ?> child) {
         children.add(child);
     }
 
+
     public void init(ProcessorContext context) {
         try {
             processor.init(context);
         } catch (Exception e) {
             throw new StreamsException(String.format("failed to initialize processor %s", name), e);
         }
+        this.time = new SystemTime();
+        this.nodeMetrics = new NodeMetrics(context.metrics(), name,  "task." + context.taskId());
+
     }
 
     public void close() {
         try {
             processor.close();
+            nodeMetrics.removeAllSensors();
         } catch (Exception e) {
             throw new StreamsException(String.format("failed to close processor %s", name), e);
         }
     }
 
     public void process(final K key, final V value) {
+        long startNs = time.nanoseconds();
```

Comment:
This is at the very critical path of streams, called millions of times per sec, so calling `time.nanoseconds()` on each call is really expensive if `DEBUG` level is used.